src/main/java/com/lineate/testyourlexicon/services/GameService.java

```diff
@@ -0,0 +1,44 @@
+package com.lineate.testyourlexicon.services;
+
+import com.lineate.testyourlexicon.dto.GameConfigurationDto;
+import com.lineate.testyourlexicon.entities.GameConfiguration;
+import com.lineate.testyourlexicon.entities.User;
+import com.lineate.testyourlexicon.exceptions.UserNotAuthenticatedException;
+import com.lineate.testyourlexicon.repositories.UserRepository;
+import com.lineate.testyourlexicon.util.GameMapper;
+import com.lineate.testyourlexicon.util.GameUtil;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class GameService {
+  private final UserRepository userRepository;
+  private final AuthenticationService authenticationService;
+
+  private void checkAuthenticated() {
+    if (!authenticationService.isAuthenticated()) {
+      throw new UserNotAuthenticatedException("User not authenticated");
+    }
+  }
+
+  public GameConfigurationDto configure(GameConfigurationDto gameConfigurationDto) {
+    checkAuthenticated();
```

Comment:
Can we check authentication on the controller level? (Protect the`/configuration` endpoint)